src/block_utils.ts

```diff
@@ -7,38 +7,37 @@ import {
     TFile,
 } from "obsidian";
 
+type CacheItem = SectionCache | ListItemCache;
+
 export abstract class BlockUtils {
     private static getBlock(
         app: App,
         editor: Editor,
         file: TFile
-    ): (SectionCache | ListItemCache) | undefined {
+    ): CacheItem | undefined {
         const cursor = editor.getCursor("to");
         const fileCache = app.metadataCache.getFileCache(file);
-
-        let currentBlock: SectionCache | ListItemCache =
-            fileCache?.sections?.find(
-                (section) =>
-                    section.position.start.line <= cursor.line &&
-                    section.position.end.line >= cursor.line
-            );
-
-        if (currentBlock.type == "list") {
-            currentBlock = fileCache.listItems?.find((list) => {
-                if (
-                    list.position.start.line <= cursor.line &&
-                    list.position.end.line >= cursor.line
-                ) {
-                    return list;
-                }
-            });
+        const sections = fileCache?.sections;
+        if (!sections || sections.length === 0) {
+            console.log('error reading FileCache (empty file?)');
+            return;
+        }
+        let currentBlock: CacheItem = (() => {
```

Comment:
Please try to create this without a self calling function.

Reply:
Okay, I will work on this @Vinzent03 , thanks